src/components/Histogram/Histogram.jsx

```diff
@@ -0,0 +1,269 @@
+   */
+  componentWillUpdate(nextProps) {
+    // regenerate the vis components if the relevant props change
+    this.visComponents = this.makeVisComponents(nextProps);
+  }
+
+  /**
+   * When the react component updates, update the d3 vis
+   */
+  componentDidUpdate() {
+    this.update();
+  }
+
+  /**
+   * Initialize the d3 chart - this is run once on mount
+   */
+  setup() {
+    const { width, height } = this.props;
+
+    // add in white background for saving as PNG
+    d3.select(this.root).append('rect')
+      .classed('chart-background', true)
+      .attr('width', width)
+      .attr('height', height)
+      .attr('x', 0)
+      .attr('y', 0)
+      .attr('fill', '#fff');
+
+    this.g = d3.select(this.root)
+      .append('g');
+
+    // add in axis groups
+    this.xAxis = this.g.append('g').classed('x-axis', true);
+    this.yAxis = this.g.append('g').classed('y-axis', true);
+
+    this.xAxis.append('line')
+      .attr('x1', 0)
+      .attr('x2', innerWidth);
+
+    // add in groups for data
+    this.bars = this.g.append('g').classed('bars-group', true);
+    this.highlightBars = this.g.append('g').classed('highlight-bars-group', true);
+
+    this.update();
+  }
+
+  makeVisComponents(props) {
+    const {
+            bins,
+            yExtent,
+            binStart,
+            binWidth,
+            width,
+            height,
+            color,
+            yFormatter,
+            innerMarginLeft = 35,
+            innerMarginRight = 10,
+            innerMarginTop = 20,
+          } = props;
+
+    const innerMargin = {
+      top: innerMarginTop,
+      right: innerMarginRight,
+      bottom: 35,
+      left: innerMarginLeft,
+    };
+
+    // Take into account the margins
+    const chartWidth = width - innerMargin.left - innerMargin.right;
+    const chartHeight = height - innerMargin.top - innerMargin.bottom;
+
+    // Convert our bin stats to actual values
+    const binEnd = (binStart) + (binWidth * (bins.length - 1));
+    const xValues = d3.range(binStart, binEnd, binWidth);
+
+    // This is a histogram, so lets use a band scale.
+    const xScale = d3.scaleBand()
+      .domain(xValues)
+      .range([0, chartWidth]);
+
+    let yDomain = yExtent;
+    if (!yDomain) {
+      yDomain = [0, d3.max(bins)];
+    }
+
+    const yScale = d3.scaleLinear().range([chartHeight, 0]).clamp(true);
+    if (yDomain) {
+      yScale.domain(yDomain);
+    }
+
+    return {
+      innerMargin,
+      binWidth,
+      bins,
+      height,
+      color,
+      chartWidth,
+      chartHeight,
+      width,
+      xValues,
+      xScale,
+      yScale,
+      yFormatter,
+    };
+  }
+
+  /**
+   * Update the d3 chart - this is the main drawing function
+   */
+  update() {
+    const { innerMargin } = this.visComponents;
+    this.g.attr('transform', `translate(${innerMargin.left} ${innerMargin.top})`);
+
+    this.updateAxes();
+    this.updateBars();
```

Comment:
I think the use of `update` is consistent with the `update` parent function. And it maintains a naming / logical distinction between "things that react creates" (`render`) and "things that are handled by d3" (`update`).